File: app/src/main/java/com/kunall17/trellassignment/Post.java
package com.kunall17.trellassignment;

import android.widget.ImageView;

import androidx.databinding.BindingAdapter;

import com.bumptech.glide.Glide;

public class Post {

    private String post;

    public Post(String post) {
        this.post = post;
    }

    public String getPostUrl() {
        return post;
    }

    public String getThumbnailUrl() {
        return post.replace("user-videos/videos/orig/", "user-images/images/orig/thumb-").replace(".mp4", ".jpg");
    }

    @BindingAdapter({"imageUrl"})
    public static void loadImage(ImageView imageView, String imagepath) {
        Glide.with(imageView.getContext()).load(imagepath).into(imageView);
    }

}

File: app/src/main/java/com/kunall17/trellassignment/viewmodels/Post.kt
package com.kunall17.trellassignment.viewmodels

import android.widget.ImageView
import androidx.databinding.BindingAdapter
import com.bumptech.glide.Glide

data class Post(val postUrl: String) {

    val thumbnailUrl: String
        get() = postUrl.replace("user-videos/videos/orig/", "user-images/images/orig/thumb-")
            .replace(".mp4", ".jpg")

    @BindingAdapter("imageUrl")
    fun loadImage(imageView: ImageView, imagepath: String?) {
        Glide.with(imageView.context).load(imagepath).into(imageView)
    }

}